static/js/settings_data.ts

```diff
@@ -112,6 +112,18 @@ export function user_can_change_logo(): boolean {
     return page_params.is_admin && page_params.zulip_plan_is_not_limited;
 }
 
+export function web_public_streams_enabled_for_realm(): boolean {
+    if (!page_params.server_web_public_streams_enabled) {
+        return false;
+    }
+
+    if (!page_params.zulip_plan_is_not_limited) {
+        return false;
+    }
+
+    return page_params.realm_enable_spectator_access;
+}
```

Comment:
Other functions in this file are mainly for checking whether a user has permission to do this task, but this is whether realm is configured as such or not, but could not find any other suitable file for this. Also, we have this function in server also but we do not include its value in `/register` response currently. Not sure whether we want to directly use that but seemed simple enough to implement in the client itself.